src/platform/frontend-metrics/metrics.js

```diff
@@ -1,13 +1,66 @@
 import Raven from 'raven-js';
 import isMetricsEnabled from './feature-flag';
 
+/**
+ * Returns 0 if Paint timing is not supported or the time of first-contentful-paint if present.
+ * @returns (number) The number retrieved from the paintEntries.startTime value, else 0.
+ */
+function contentfulPaintEntry() {
+  const paintEntries = performance.getEntriesByName('first-contentful-paint');
+  // If browser doesn't support first-contentful-paint, we set the value to 0
+  const firstPaint = typeof paintEntries === 'undefined' ? 0 : paintEntries[0].startTime;
+
+  return firstPaint;
+}
+
+/**
+ * Returns a formatted metrics payload FormData object to send to backend
+ * @param (PerformanceNavigationTiming) An object with information from the browser's Performance API
+ * @returns (FormData) A new FormData object with an array of metrics and page_id parameters
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry
+ */
+function buildMetricsPayload(entry) {
+  const totalPageLoad = entry.duration;
+  const firstByte = entry.responseStart;
+  const domProcessing = entry.domComplete - entry.domInteractive;
+  const domComplete = entry.domComplete - entry.requestStart;
+  const domInteractive = entry.domInteractive;
+  const firstPaint = contentfulPaintEntry();
+
+  const metrics = {
+    totalPageLoad,
+    firstByte,
+    domProcessing,
+    domComplete,
+    domInteractive,
+    firstPaint,
+  };
+
+  const metricsData = new FormData();
+  const metricsArray = [];
+  const pageUrl = window.location.pathname;
+
+  Object.keys(metrics).forEach(metric => {
+    metricsArray.push({
+      metric,
+      duration: metrics[metric]
+    });
+  });
+
+  metricsData.append('metrics', JSON.stringify(metricsArray));
+  metricsData.append('page_id', pageUrl);
+
+  return metricsData;
```

Comment:
This should return a JSON object if at all possible so the backend doesn't have to parse it.